### PR TITLE
Rework file input

### PR DESCRIPTION
**Pull request by jt-traub**
This changes the existing file input to use std::istream throughout the bulk of the code rather than Aorderfile and Ainputfile. Output is still done via the custom file classes, but those will be changed eventually. as well.

### CODE REVIEW

**Comment by valdisz** [comment]
As all tests and checks are passing, it means it works

File: aregion.cpp [hunk 1]
```diff
@@ -103,11 +103,13 @@ TownInfo::~TownInfo()
 	if (name) delete name;
 }
 
-void TownInfo::Readin(Ainfile *f, ATL_VER &v)
+void TownInfo::Readin(istream &f)
```
Comment (valdisz) [inline aregion.cpp:106]: As I understand `ATL_VER &v` was not used at all...

**Review by valdisz — APPROVED** [review]
As the tests showed, output before and after does not change.

**Review by artyomtrityak — APPROVED** [review]

File: aregion.cpp (hunk 1 above)
Comment (jt-traub) [reply aregion.cpp:106]: Correct.  It had been passed down to some but not all functions, and when I chased through, I found that it was not *actually* used anywhere in the reading functions *except* in the very top level to verify that the version was compatable, so I just got rid of it all the way up the call chain(s) from anything below that top level.

**Pull request merged**